### PR TITLE
DB-8980 Unit test SICompactionState.mutate

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: pom.xml [hunk 1]
```diff
@@ -121,6 +121,12 @@
             <version>1.9.5</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-library</artifactId>
+            <version>1.3</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (OlegMazurov) [inline pom.xml]: This is a duplicate and most likely redundant dependency at the topmost level.

Comment (OlegMazurov) [reply pom.xml]: If you want _hamcrest-library_ to be available to all modules for unit tests, this is the right location to specify the dependency (along with junit itself). _hamcrest-core_ should also be moved here then.
The _dependencyManagement_ section is good when we want to specify a dependency selectively in a subset of modules but manage the attributes of that dependency (version, scope) in one place.
If _hamcrest_ dependencies are specified as global (spliceengine-parent dependencies) there is no need to add them to the _dependencyManagement_ section.

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh5.12.2

File: pom.xml (hunk 1 above)
Comment (arnaud-splice) [reply pom.xml]: I removed the hamcrest reference in _dependencyManagement_ and added hamcrest (without -core or -library) along with junit.

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    private final boolean purgeDeletedRows;
+    private long maxTombstoneTimestamp;
+
+    SICompactionStateMutate(boolean purgeDeletedRows) {
+        this.purgeDeletedRows = purgeDeletedRows;
+        this.dataToReturn = new TreeSet<>(KeyValue.COMPARATOR);
+        this.maxTombstoneTimestamp = 0;
+    }
+
+    private boolean isSorted(List<Cell> list) {
```
Comment (dgomezferro) [inline hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java]: You could use Guava here: `Ordering.from(KeyValue.COMPARATOR).isOrdered(list);`

File: hbase_storage/src/test/java/com/splicemachine/si/impl/server/SICompactionStateMutateTest.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        assertThat(outputCells, equalTo(Arrays.asList(
+                SITestUtils.getMockCommitCell(0x300),
+                SITestUtils.getMockAntiTombstoneCell(0x300),
+                SITestUtils.getMockValueCell(0x300))));
+    }
+
+    @Test
+    public void committed() {
+        TxnView committedWithRootParent = TxnTestUtils.getMockCommittedTxn(0x100, 0x200);
+        assertTrue(SICompactionStateMutate.committed(committedWithRootParent));
```
Comment (dgomezferro) [inline hbase_storage/src/test/java/com/splicemachine/si/impl/server/SICompactionStateMutateTest.java]: Can you rename `SICompactionStateMutate.committed` to `isCommitted` ? I think that would be clearer.

File: hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java (hunk 2 above)
Comment (arnaud-splice) [reply hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java]: I wanted to, but I could not make it work without adding a new dependency and was wondering if it was worth it.

File: hbase_storage/src/test/java/com/splicemachine/si/impl/server/SICompactionStateMutateTest.java (hunk 3 above)
Comment (arnaud-splice) [reply hbase_storage/src/test/java/com/splicemachine/si/impl/server/SICompactionStateMutateTest.java]: done

**Pull request merged**